src/backend/go.rs

```diff
@@ -63,21 +65,29 @@ impl Backend for GoBackend {
         let settings = Settings::get();
         settings.ensure_experimental("go backend")?;
 
-        // if the (semantic) version has no v prefix, add it
-        // we allow max. 6 digits for the major version to prevent clashes with Git commit hashes
-        let version = if regex!(r"^\d{1,6}(\.\d+)*([+-.].+)?$").is_match(&ctx.tv.version) {
-            format!("v{}", ctx.tv.version)
-        } else {
-            ctx.tv.version.clone()
-        };
+        if ctx.tv.version.starts_with("v") {
+            bail!("versions should not start with a 'v', please remove it");
+        }
 
-        CmdLineRunner::new("go")
+        let result = CmdLineRunner::new("go")
             .arg("install")
-            .arg(format!("{}@{}", self.name(), version))
+            .arg(format!("{}@v{}", self.name(), ctx.tv.version.clone()))
```

Comment:
```suggestion
            .arg(format!("{}@v{}", self.name(), &ctx.tv.version))
```

I know this is nitpicky but it bugs me